pkg/reconciler/kubernetes/tektonresult/transform.go

```diff
@@ -210,3 +226,70 @@ func updateApiConfig(p interface{}) mf.Transformer {
 		return nil
 	}
 }
+
+// updates env keys with the secret name into "tekton-results-api" deployment in "api" container
+func updateEnvWithSecretName(props v1alpha1.ResultsAPIProperties) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if props.SecretName == "" || u.GetKind() != "Deployment" || u.GetName() != deploymentAPI {
+			return nil
+		}
+
+		dep := &appsv1.Deployment{}
+		err := k8sruntime.DefaultUnstructuredConverter.FromUnstructured(u.Object, dep)
+		if err != nil {
+			return err
+		}
+
+		// find the matching container and add env and secret name object
+		for containerIndex, container := range dep.Spec.Template.Spec.Containers {
+			if container.Name != apiContainerName {
+				continue
+			}
+
+			// get existing env from the container
+			existingEnv := container.Env
+			if existingEnv == nil {
+				existingEnv = make([]corev1.EnvVar, 0)
+			}
+
+			// update only allowed properties
+			for _, propertyKey := range allowedPropertySecretKeys {
+				newEnv := corev1.EnvVar{
+					Name: propertyKey,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: props.SecretName,
+							},
+							Key:      propertyKey,
```

Comment:
Will there be any issues assume if a key is missed in secret ?

Reply:
@piyush-garg I have added `Optional: ptr.Bool(true)`, So it will not give any issue, if the key not present in the secret.

Reply:
ohk, perfect